### ISSUE TITLE
possible bug: bare git without packed-refs file wouldn't show up in the scan result

### ISSUE THREAD

**Issue by darkclip** [w4/rgit#50]
if you init a bare repo in the scan path, it won't create "packed-refs" file by default, in this case, the program wouldn't recognize it as a repo. you have to use "git pack-refs --all" command to create the file for the repo to be recognized by rgit.

**Comment by stappersg** [comment]
_Nice_.

I'm now testing
```patch
--- a/server_side/rgitsss
+++ b/server_side/rgitsss
@@ -111,6 +111,8 @@ rgitsss_init() {
     COMMAND="${COMMAND}git init --bare --initial-branch=main ${BASE}"
     su ${ACCOUNT} --shell=/usr/bin/bash \
         --command="${COMMAND}"
+    su ${ACCOUNT} --shell=/usr/bin/bash \
+        --command="cd ${DIR}/${BASE} && git pack-refs --all"
 }
 
 rgitsss_describe() {
```

**Comment by w4** [comment]
Interesting. I wonder if there's any other identifying file in the repo that can be used instead. Maybe `git-daemon-export-ok` which is the file that needs to be created to allows clones?

**Comment by stappersg** [comment]
![afbeelding](https://github.com/w4/rgit/assets/35434094/b46fd57c-67f5-4ac4-b4de-f228f7e79079)

The `git pack-refs --all`  does provide what `rgit` seams to need for **display**ing.


For _"allow to be cloned"_ has script `rgitsss` subcommand `publish`, which does [`touch git-daemon-export-ok`](https://github.com/w4/rgit/pull/29/files#diff-e84638e3f7fd85d2b5288dbbb2f4af6661a5188aee026034615543db9f3e95b6R177).

**Comment by darkclip** [comment]
@w4 I think it doesn't matter that much what kind of identifier to be used, as long as you give it a proper documentation.

**Comment by stappersg** [comment]
On Tue, Jan 23, 2024 at 08:04:06AM -0800, DC wrote:
> Closed #50 as completed.

And thanks for the reminder on https://github.com/w4/rgit/pull/29
> 

Groeten
Geert Stappers
-- 
Silence is hard to parse
